sourcegraph/cody
# feat(audoedit): enable E2E UX testing with mock responses

- Enables us to use mocked autoedit predictions defined in the comment template as API responses. Allows offline testing and development of the full autoedit flow, from trigger events to rendering and acceptance. 
- Part of [CODY-4412: Implement experimental inline renderer](https://linear.app/sourcegraph/issue/CODY-4412/implement-experimental-inline-renderer)
- Closes [CODY-4451: Testing: enable E2E flow testing with mock responses in comment templates](https://linear.app/sourcegraph/issue/CODY-4451/testing-enable-e2e-flow-testing-with-mock-responses-in-comment)
- Built on top of https://github.com/sourcegraph/cody/pull/6235

## Test plan

- CI
- Manually tested:
  - Add `"cody.experimental.autoedits.use-mock-responses": true,` to your VS Code settings
  - Go to autoedit rendering testing examples in [the cody-chat-eval repo](https://github.com/sourcegraph/cody-chat-eval/blob/main/code-matching-eval/edits_experiments/examples/renderer-testing-examples).
  - Disable internet
  - Navigate through renderer testing examples and see suggestions being displayed.
